src/udev/udevd.c

```diff
@@ -854,9 +854,21 @@ static int event_run(Event *event) {
         return 1; /* event is now processing. */
 }
 
+static bool devpath_conflict(const char *a, const char *b) {
+        /* This returns true when two paths are equivalent, or one is a child of another. */
+
+        if (!a || !b)
+                return false;
+
+        for (; *a != '\0' && *b != '\0'; a++, b++)
+                if (*a != *b)
+                        return false;
+
+        return *a == '/' || *b == '/' || *a == *b;
+}
```

Comment:
could use a test case I think. given this is not obvious

Reply:
hmm, *a == *b only holds if both are NUL, right?

Reply:
> hmm, *a == *b only holds if both are NUL, right?

Right.

Reply:
Added testcases.